### PR TITLE
Merge `v0.37.0_release` into `master`

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Included changes from the release branch into master. 

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-68100]

### CODE REVIEW

**Review by AmintorDusko — APPROVED** [review]
Looks good!

**Review by multiphaseCFD — COMMENTED** [review]
Thanks @vincentmr! Just a few question.

File: docker/Dockerfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ ARG DEBIAN_FRONTEND=noninteractive
 ARG GCC_VERSION=11
 ARG LIGHTNING_VERSION=v0.36.0
 ARG PENNYLANE_VERSION=v0.36.0
-ARG ROCM_INSTALLER=https://repo.radeon.com/amdgpu-install/6.1.1/ubuntu/jammy/amdgpu-install_6.1.60101-1_all.deb
+ARG ROCM_INSTALLER=https://repo.radeon.com/amdgpu-install/5.7/ubuntu/jammy/amdgpu-install_5.7.50700-1_all.deb
```
Comment (multiphaseCFD) [inline docker/Dockerfile:26]: Is there any reason to downgrade rocm version?

Comment (vincentmr) [reply docker/Dockerfile:26]: Didn't match the image version [here](https://github.com/PennyLaneAI/pennylane-lightning/blob/be9459f489d32d0624c37c71730e7f92379f5ba4/docker/Dockerfile#L196) with which it was tested on LUMI.

**Review by multiphaseCFD — APPROVED** [review]

**Pull request merged**